### PR TITLE
Add support for TypeScript 5.0

### PR DESCRIPTION
**Pull request by wgoehrig**
Needed for https://github.com/iTwin/itwinjs-core/pull/5242

### CODE REVIEW

File: dist/configs/itwinjs-recommended.js [hunk 1]
```diff
@@ -129,6 +129,20 @@ module.exports = {
         "selector": "enumMember",
         "format": null,
         "leadingUnderscore": "allow"
+      },
+      {
+        "selector": [
```
Comment (paulius-valiunas) [inline dist/configs/itwinjs-recommended.js:134]: do we have to list all these selectors here, or would `default` be good enough?

Comment (wgoehrig) [reply dist/configs/itwinjs-recommended.js:134]: I just tried default, but it doesn't seem to work... TBH, I copied this straight from https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/naming-convention.md#ignore-properties-that-require-quotes

Comment (paulius-valiunas) [reply dist/configs/itwinjs-recommended.js:134]: ok, I will trust their documentation then

File: package.json [hunk 2]
```diff
@@ -52,9 +52,9 @@
     "@types/eslint": "~7.2.13",
     "@types/estree": "~0.0.48",
     "@types/node": "^18.11.5",
-    "@typescript-eslint/typescript-estree": "~4.31.0",
+    "@typescript-eslint/typescript-estree": "~5.55.0",
     "eslint": "^7.11.0",
     "mocha": "^10.0.0",
-    "typescript": "~4.4.0"
+    "typescript": "~5.0.2"
```
[the remark this answers is not in the document]
Comment (paulius-valiunas) [reply package.json:58]: and add a changeset since this seems like an important change

[the remark this answers is not in the document]
Comment (wgoehrig) [reply package.json:58]: Just updated the peerDependency, but going to defer to @yashincontrol and @aruniverse for the changeset.

[the remark this answers is not in the document]
Comment (aruniverse) [reply package.json:58]: just an update, switching to beachball for publishing. once thats merged we'll update this pr and get this published tomorrow

File: change/@itwin-eslint-plugin-d973f288-c15d-4b29-8046-cd4e3cc61d8d.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
+  "comment": "Added support for TypeScript 5.0",
+  "packageName": "@itwin/eslint-plugin",
+  "email": "yashincontrol@gmail.com",
```
Comment (aruniverse) [inline change/@itwin-eslint-plugin-d973f288-c15d-4b29-8046-cd4e3cc61d8d.json:5]: i would recommend not exposing your personal/private email.
use your gh account's noreply email address

File: package.json [hunk 4]
```diff
@@ -48,17 +48,17 @@
     "require-dir": "^1.2.0"
   },
   "peerDependencies": {
-    "eslint": "^7.0.0",
-    "typescript": "^3.7.0 || ^4.0.0"
+    "eslint": "^8.36.0",
```
Comment (aruniverse) [inline package.json:51]: i didnt realize we had merged the eslint@8 pr. need to run nightman to check if this works

File: change/@itwin-eslint-plugin-d973f288-c15d-4b29-8046-cd4e3cc61d8d.json (hunk 3 above)
Comment (aruniverse) [reply change/@itwin-eslint-plugin-d973f288-c15d-4b29-8046-cd4e3cc61d8d.json:5]: ive created a separate issue to address this, https://github.com/iTwin/eslint-plugin/issues/9.
but for this pr, just change the email in here manually

File: package.json (hunk 4 above)
Comment (yashincontrol) [reply package.json:51]: It's not from Master. I only pulled the beachball config files from master

Comment (yashincontrol) [reply package.json:51]: The update was done by Bill. While resolving merge conflicts it somehow got messed up. So I updated to eslint8 again and generated the lock file

Comment (yashincontrol) [reply package.json:51]: Weird, I'm pretty sure it was 8.36.0 before I merged master. @wgoehrig could you verify please?

Comment (aruniverse) [reply package.json:51]: @yashincontrol  you can verify by looking through the history 
https://github.com/iTwin/eslint-plugin/commits/ts5/package.json

**Review by aruniverse — APPROVED** [review]

**Pull request merged**